cmd/thanos/query.go

```diff
@@ -464,8 +468,10 @@ func runQuery(
 
 		g.Add(func() error {
 			level.Info(logger).Log("msg", "Listening for StoreAPI gRPC", "address", grpcBindAddr)
+			readinessProber.SetReady()
```

Comment:
Why in this case are we setting Ready? I'm a little confused, because in other cases we are setting Healthy before we start serving http.

Is there a pattern?

Reply:
Pattern was discussed in the comment